### PR TITLE
TW-1127: preview multiple files web

### PR DESCRIPTION
**Pull request by sherlockvn**
### Demo web with add button:

https://github.com/linagora/twake-on-matrix/assets/43041967/387fefd5-3ffb-4ed6-a34a-dafd8a94661f

### Demo web with drag and drop: 

https://github.com/linagora/twake-on-matrix/assets/43041967/8a7e430e-ec6a-4b91-b283-d4cc10f51c50

### CODE REVIEW

File: assets/images/ic_file_video.svg
Comment (hoangdat) [inline assets/images/ic_file_video.svg:1]: can you show me the link to these icons?

File: lib/di/global/get_it_initializer.dart [hunk 1]
```diff
@@ -268,5 +269,8 @@ class GetItInitializer {
     getIt.registerLazySingleton<UpdatePinnedMessagesInteractor>(
       () => UpdatePinnedMessagesInteractor(),
     );
+    getIt.registerFactory<SendFilesOnWebWithCaptionInteractor>(
+      () => SendFilesOnWebWithCaptionInteractor(),
```
Comment (hoangdat) [inline lib/di/global/get_it_initializer.dart:273]: how about the problem which you said about the order of the message?

File: assets/images/ic_file_video.svg
Comment (sherlockvn) [reply assets/images/ic_file_video.svg:1]: here: https://www.figma.com/file/uLw3iWnOEGaRlJnfPz4Pv0/Twake-%5BMatrix%5D?type=design&node-id=6497-117123&mode=dev

File: lib/di/global/get_it_initializer.dart (hunk 1 above)
Comment (sherlockvn) [reply lib/di/global/get_it_initializer.dart:273]: i have tried both `registerFactory` and `registerSingleton`. But, the order of messages is not correct, so i think we can do it in another ticket

File: lib/pages/chat_draft/draft_chat.dart [hunk 2]
```diff
@@ -298,14 +298,14 @@ class DraftChatController extends State<DraftChat>
...
 
-  @override
   void sendFileOnWebAction(
     BuildContext context, {
     Room? room,
   }) async {
     final sendFileOnWebInteractor = getIt.get<SendFileOnWebInteractor>();
     final result = await FilePicker.platform.pickFiles(
       withData: true,
+      allowMultiple: true,
```
Comment (hoangdat) [inline lib/pages/chat_draft/draft_chat.dart:308]: please remove it, we will do it later

File: lib/pages/chat/send_file_dialog_style.dart [hunk 3]
```diff
@@ -6,7 +6,7 @@ class SendFileDialogStyle {
 
   static const double dialogWidth = 328;
```
Comment (sherlockvn) [inline lib/pages/chat/send_file_dialog_style.dart:7]: 360

File: lib/pages/chat/send_file_dialog_view.dart [hunk 4]
```diff
@@ -143,3 +128,87 @@ class SendFileDialogView extends StatelessWidget {
...
+      },
+      child: SizedBox(
+        width: SendFileDialogStyle.imageSize,
+        height: SendFileDialogStyle.imageSize,
+        child: ClipRRect(
+          borderRadius: BorderRadius.circular(
+            SendFileDialogStyle.imageBorderRadius,
+          ),
+          child: Image.memory(
+            files.first.bytes!,
```
Comment (nqhhdev) [inline lib/pages/chat/send_file_dialog_view.dart:163]: Should check `files.first.bytes == null`

**Review by nqhhdev — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]
please rebase

**Pull request merged**